theseus/core/cost_function.py

```diff
@@ -94,7 +94,8 @@ def __call__(
 class AutoDiffCostFunction(CostFunction):
     def __init__(
         self,
-        optim_vars: List[Variable],
+        # optim_vars: List[Variable],
+        optim_vars: list,  # TODO: List[Vector] raises mypy error?
```

Comment:
@luisenp without this update the mypy test was giving an error since it doesn't seem to accept `Vector` even though it's a child of `Variable`, do you know if there's a better update to this?

```
mypy.....................................................................Failed                                        
- hook id: mypy                                                                                                        
- exit code: 1                                                                                                         
                                                           
theseus/optimizer/nonlinear/tests/test_backwards.py:43: error: Argument 1 to "AutoDiffCostFunction" has incompatible ty
pe "List[Vector]"; expected "List[Variable]"  [arg-type]                                                               
theseus/optimizer/nonlinear/tests/test_backwards.py:43: note: "List" is invariant -- see https://mypy.readthedocs.io/en
/stable/common_issues.html#variance                        
theseus/optimizer/nonlinear/tests/test_backwards.py:43: note: Consider using "Sequence" instead, which is covariant    
examples/backward_modes.py:51: error: Argument 1 to "AutoDiffCostFunction" has incompatible type "List[Vector]"; expect
ed "List[Variable]"  [arg-type]
examples/backward_modes.py:51: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.htm
l#variance
examples/backward_modes.py:51: note: Consider using "Sequence" instead, which is covariant
Found 2 errors in 2 files (checked 101 source files)
```

Reply:
Any reason we want to restrict to `Vector` only, here?

Reply:
I've reverted this to `List[Variable]` and added ignores in the files causing this issue for now (`examples/backward_modes.py` and `theseus/optimizer/nonlinear/tests/test_backwards.py`). It seems like the issue is that it doesn't recognize a list of `Vector` objects as a subclass of a `Variable` list?